scripts/gitlab_runner_pre_build/pre_build.py

```diff
@@ -48,6 +48,42 @@ def _token_to_sts_request(raw_jwt, decoded_jwt):
     return assume_role_kwargs
 
 
+def _durable_assume_role_request(assume_role_kwargs):
+    attempts = 0
+    max_attempts = 5
+
+    while attempts < max_attempts:
+        try:
+            req = urllib.request.Request(
+                "https://sts.amazonaws.com/?Action=AssumeRoleWithWebIdentity&Version=2011-06-15&"
+                + urllib.parse.urlencode(assume_role_kwargs),
+                headers={"Accept": "application/json"},
+            )
+
+            with urllib.request.urlopen(req) as response:
+                if response.getcode() == 200:
+                    response = json.loads(response.read().decode("utf-8"))
+                    return response
+                else:
+                    raise ValueError(
```

Comment:
Is this new check based on some failure mode you've seen recently?

Reply:
It's based on being overlay paranoid. AWS doesn't enumerate the non-error statuses possible so I would rather explicitly catch it.